p2p/address-book/src/lib.rs

```diff
@@ -82,3 +81,21 @@ pub async fn init_address_book<Z: NetworkZone>(
 
     Ok(address_book)
 }
+
+use sealed::*;
+mod sealed {
+    use super::*;
+
+    /// An internal trait for the address book for a [`NetworkZone`] that adds the requirement of [`borsh`] traits
+    /// onto the network address.
+    pub trait BorshNetworkZone: NetworkZone<Addr = Self::BorshAddr> {
+        type BorshAddr: NetZoneAddress + borsh::BorshDeserialize + borsh::BorshSerialize;
+    }
```

Comment:
I moved the borsh requirement to the address book instead of feature flagging it on `cuprate-p2p-core`, I think this is a lot clearer.

Reply:
Just to reduce type noise in the address book